flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/functions/utils/UserDefinedFunctionUtils.scala

```diff
@@ -728,10 +727,12 @@ object UserDefinedFunctionUtils {
         expected.isPrimitive && Primitives.wrap(expected) == candidate ||
         candidate == classOf[Date] && (expected == classOf[Int] || expected == classOf[JInt])  ||
         candidate == classOf[Time] && (expected == classOf[Int] || expected == classOf[JInt]) ||
-        candidate == classOf[Timestamp] && (expected == classOf[Long] ||
-            expected == classOf[JLong]) ||
```

Comment:
I think this is a big incompatible change which should be discussed publicly. Because we encourage users to use long to represent timestamp in UDFs in [documentation](https://ci.apache.org/projects/flink/flink-docs-master/dev/table/functions/udfs.html#scalar-functions). And many users are using in this way, this will affect many users. 
IIUC, this pull request doesn't blocked by this change, and can be done in another JIRA issue.

Reply:
Decide to disable this in the offline discussing @KurtYoung @JingsongLi